src/babel.js

```diff
@@ -295,16 +295,15 @@ export default function ({types: t}) {
 
           // we replace styles with the function call
           const [id, css, loc] = state.styles.shift()
-
-          if (isGlobal) {
-            path.replaceWith(makeStyledJsxTag(id, css.source || css, css.modified))
-            return
-          }
-
-          const useSourceMaps = Boolean(state.file.opts.sourceMaps)
           let transformedCss
+          const useSourceMaps = Boolean(state.file.opts.sourceMaps)
 
-          if (useSourceMaps) {
+          if (isGlobal) {
+            transformedCss = transform(
+              null,
+              (css.modified || css)
+            )
+          } else if (useSourceMaps) {
```

Comment:
We should have source maps for global styles too I think.

Maybe you can do something like this:

```js
if (useSourceMaps) {
   transformedCss = [
       transform(
          isGlobal ? null : String(state.jsxId),
          // ...
       ),
       // ...
   ]
} else {
      transformedCss = transform(
         isGlobal ? null : String(state.jsxId),
         // ...
      )
}
```